lib/desktop_screens_new/history_screen/widgets/history_list_tile.dart

```diff
@@ -30,225 +42,296 @@ class HistoryCardWidget extends StatefulWidget {
+                        child: Wrap(
+                          alignment: WrapAlignment.center,
+                          spacing: 5,
+                          runSpacing: 5,
+                          children: [
+                            ...widget.tags.map((tag) {
+                              return Container(
+                                decoration: BoxDecoration(
+                                  color: ColorConstants.MILD_GREY,
+                                  borderRadius: BorderRadius.circular(10),
                                 ),
-                              ),
-                            );
-                          }).toList(),
-                        ],
+                                padding: EdgeInsets.symmetric(
+                                    vertical: 2, horizontal: 8),
+                                child: Text(
+                                  tag.text,
+                                  style: TextStyle(
+                                    color: Color(0xFF909090),
+                                    fontSize: 10,
+                                  ),
+                                ),
+                              );
+                            }).toList(),
+                          ],
+                        ),
                       ),
-                    ),
-                    Expanded(
-                      flex: 1,
-                      child: Row(
-                        mainAxisAlignment: MainAxisAlignment.center,
-                        children: [
-                          InkWell(
-                            onTap: () {
-                              openFileReceiptBottomSheet();
-                            },
-                            child: Icon(
-                              Icons.done_all,
-                              size: 14,
-                              color: Color(0xFF909090),
-                            ),
-                          ),
-                        ],
+                      Expanded(
+                        flex: 1,
+                        child: Row(
+                          mainAxisAlignment: MainAxisAlignment.center,
+                          children: [
+                            context.watch<HistoryProvider>().typeSelected ==
+                                    HistoryType.send
+                                ? InkWell(
+                                    onTap: () {
+                                      openFileReceiptBottomSheet();
+                                    },
+                                    child: Icon(
+                                      Icons.done_all,
+                                      size: 14,
+                                      color: Color(0xFF909090),
+                                    ),
+                                  )
+                                : Consumer<FileProgressProvider>(
+                                    builder: (_c, provider, _) {
```

Comment:
Can we add this builder into another function/ component ?
It has become unreadable.

Reply:
@sachins-geekyants Done